_posts/2018-05-29-Introducing-Bazel-Cquery.md

```diff
@@ -0,0 +1,90 @@
+---
+layout: posts
+title: Introducing Bazel Cquery
+---
+
+(If you’re already familiar with bazel’s `query` command, skip ahead to the section titled ‘Introducing Cquery’.)
+
+## Querying Your Build
+Have you ever wondered why making a certain change causes Bazel to rebuild a seemingly unrelated binary? Have you ever wanted to run a particular subset of tests based on some shared attribute value but didn’t know how to gather them? Or maybe you have targets that work across a variety of platforms and you want to make sure none of them rely on incompatible dependencies? Bazel’s query commands can help remedy such issues.
+
+Bazel uses the [Bazel Query Language](https://docs.bazel.build/versions/master/query.html) to articulate questions such as those listed above. This language includes a set of [functions](https://docs.bazel.build/versions/master/query.html), such as ‘deps’ and ‘somepath’, which can be combined and nested to describe more complex queries. 
+
+The query commands also support a set of flag-based options that govern query behavior. For example, [--noimplicit_deps](https://docs.bazel.build/versions/master/query.html#implicit_deps) limits results to only include targets that are explicitly declared in a BUILD file. The flags also support a selection of [output formats](https://docs.bazel.build/versions/master/query.html#output-formats) for the results of the query, such as a [graph output](https://docs.bazel.build/versions/master/query.html#output-graph) (especially helpful for visualizing the collection of paths between two nodes).
+
+## Introducing `bazel cquery`
+Bazel has long supported the `query` command. We’re excited to announce a second* command, `cquery`!
+
+What’s the difference? `query` doesn’t understand build flags and returns all possible answers to a given query expression. `cquery` (configurable query) runs at a later point in the build process, after flag evaluation and [configurable attribute](https://docs.bazel.build/versions/master/be/common-definitions.html#configurable-attributes) resolution. Thus, it can understand [build options](https://docs.bazel.build/versions/master/command-line-reference.html#command-line-reference) and give the answer to a specific Bazel invocation as dictated by its set of flags. Since it runs at a later point, ‘cquery’ is by nature slower than ‘query’.  
+
+`cquery` has yet to reach feature parity with `query`, but as we develop it further, `cquery` will also be able to expose information to which `query` does not have access. See the comparison below for what’s currently supported. 
+
+|                      | `query`                                                                    |`cquery`                              | both                                                                            |
```

Comment:
Can you please use a HTML table instead? Feel free to reuse the code in the previous PR: https://github.com/bazelbuild/bazel-blog/pull/81

Reply:
Why? I fixed our setup to also parse tables, and added some templating to make them look nicer. See https://blog.bazel.build/2018/04/13/preliminary-sandboxfs-support.html for a rendered example.

Reply:
Oh nice! I didn't realize you fixed it already.